### PR TITLE
Doc updates hpx build system v1

### PR DESCRIPTION
**Pull request by rstobaugh**
Updates to documentation; grammar updates.

### CODE REVIEW

File: docs/sphinx/manual/building_hpx.rst [hunk 1]
```diff
@@ -998,66 +996,65 @@ How to build HPX under Windows 10 x64 with Visual Studio 2015
...
-    should be installed to.
+  * ``CMAKE_INSTALL_PREFIX`` the |HPX| root directory where the future builds of |HPX|
+    should be installed.
 
     .. note::
 
-       HPX is a BIG software collection and I really don't recommend using the
-       default ``C:\Program Files\hpx``. I prefer simpler paths *without* white
-       space, like ``C:\bin\hpx`` or ``D:\bin\hpx`` etc.
+       |HPX| is a very large software collection, so it is not recommend to use the
```
Comment (msimberg) [inline docs/sphinx/manual/building_hpx.rst]:
```suggestion
       |HPX| is a very large software collection, so it is not recommended to use the
```

File: cmake/templates/cmake_toolchains.rst.in [hunk 2]
```diff
@@ -14,13 +14,13 @@
 |cmake| toolchains shipped with |hpx|
 =====================================
 
-In order to compile HPX for various platforms, we provide a variety of toolchain
-files that take care of setting up various |cmake| variables like compilers etc.
+In order to compile |HPX| for various platforms, we provide a variety of toolchain
```
Comment (msimberg) [inline cmake/templates/cmake_toolchains.rst.in]:
```suggestion
In order to compile |hpx| for various platforms, we provide a variety of toolchain
```

File: cmake/templates/cmake_variables.rst.in [hunk 3]
```diff
@@ -31,43 +31,43 @@ The options are split into these categories:
...
 .. option:: BOOST_ROOT:PATH
 
-   Specifies where to look for the |boost|_ installation to be used for
-   compiling |hpx| Set this if CMake is not able to locate a suitable version of
-   |boost|_ The directory specified here can be either the root of a installed
-   Boost distribution or the directory where you unpacked and built |boost|_
+   Specifies where to look for the Boost installation to be used for
+   compiling |hpx|. Set this if CMake is not able to locate a suitable version of
+   Boost. The directory specified here can be either the root of an installed
+   Boost distribution or the directory where you unpacked and built Boost
```
Comment (msimberg) [inline cmake/templates/cmake_variables.rst.in]: I just want to check that the `|boost|_` and `|cmake|_` to `Boost` and `CMake` replacements are intentional. `|boost|` will be rendered as `Boost`, and `|boost|_` will be rendered as as a clickable link with the text `Boost`.

File: cmake/templates/cmake_toolchains.rst.in (hunk 2 above)
Comment (msimberg) [reply cmake/templates/cmake_toolchains.rst.in]: This needs to be lower case to render as italic *HPX*.

File: cmake/templates/cmake_variables.rst.in (hunk 3 above)
Comment (rstobaugh) [reply cmake/templates/cmake_variables.rst.in]: It's intentional. The use of hyperlinks is inconsistent throughout the manual, so the others and I decided to only include external links the first time a term is mentioned, and to include internal links within the manual itself every time a term is mentioned.

File: cmake/templates/cmake_toolchains.rst.in (hunk 2 above)
Comment (rstobaugh) [reply cmake/templates/cmake_toolchains.rst.in]: I'll go back and change it--thanks!

File: cmake/templates/cmake_variables.rst.in (hunk 3 above)
Comment (msimberg) [reply cmake/templates/cmake_variables.rst.in]: Sounds good, thanks!

**Comment by msimberg** [comment]
@rstobaugh I committed a small fix to avoid unnecessary back-and-forth. I'll go ahead and merge this now. Keep them coming!

**Comment by hkaiser** [comment]
@rstobaugh Thanks for your contributions. Please get in contact with @aserio if you are interested in receiving a STE||AR-Group t-shirt. We give that away to our first-time contributors.

**Pull request merged**